amigo-lib/src/main/java/me/ele/amigo/AmigoCallback.java

```diff
@@ -0,0 +1,77 @@
+
+package me.ele.amigo;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ActivityInfo;
+import android.os.Handler;
+import android.os.Message;
+import android.util.Log;
+
+import me.ele.amigo.reflect.FieldUtils;
+import me.ele.amigo.utils.ComponentUtils;
+
+import static me.ele.amigo.AmigoInstrumentation.EXTRA_TARGET_INFO;
+import static me.ele.amigo.AmigoInstrumentation.EXTRA_TARGET_INTENT;
+
+public class AmigoCallback implements Handler.Callback {
+
+    private static final String TAG = AmigoCallback.class.getSimpleName();
+
+    public static final int LAUNCH_ACTIVITY = 100;
+
+
+    private Handler.Callback mCallback = null;
+    private Context context;
+
+
+    public AmigoCallback(Context context, Handler.Callback callback) {
+        mCallback = callback;
+        this.context = context;
+    }
+
+    @Override
+    public boolean handleMessage(Message msg) {
+
+        if (msg.what == LAUNCH_ACTIVITY) {
+            return handleLaunchActivity(msg);
+        }
+        if (mCallback != null) {
```

Comment:
what you mean "delegate to mCallback.handleMessage everytime" ?
 AmigoCallback is the mCallback exactly.